### ISSUE TITLE
If the running environment does not support Windows and navigator, the two objects cannot be used.

### ISSUE THREAD

**Issue by chenqunjian** [travist/jsencrypt#123]
If the system does not support two objects, Windows and navigator, this js will not be used, and it was recently developed using this encryption in WeChat small program to find this hole.

**Comment by sept08** [comment]
yeah, I also encountered similar problem as I developing a Wechat applet.

**Comment by JulienParis** [comment]
I also encountered this problem : when I'm running jsencrypt in Safari there are no problem, when I'm running it with Firefox the message "navigator is not defined" appears and stops my scripts from running... (stack : nuxt/vue.js, import of jsencrypt as a nuxt plugin or in any function)

**Comment by qzmer1104** [comment]
same problem。

**Comment by monkeylord** [comment]
How about
> window=this

**Comment by jcs12311** [comment]
把https://github.com/travist/jsencrypt/blob/master/bin/jsencrypt.js 源码拷贝到小程序目录，增加以下代码到源码，测试了是可以加密的。
`var navigator = {
    appName: 'Netscape',
    userAgent: '',
};
var window = {};`

**Comment by travist** [comment]
Try with 3.4.0
